### PR TITLE
Add CloudWatch config to HCP Log Streaming Destination

### PR DESCRIPTION
**Pull request by leahrob**
Adding `cloudwatch` as a config option for the `HCP Log Streaming Destination` resource.  [IFACE-2483](https://hashicorp.atlassian.net/browse/IFACE-2483)

### :building_construction: Acceptance tests

- [x] Are there any feature flags that are required to use this functionality?
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

Output from acceptance testing:

<details>
  <summary>TestAccHCPLogStreamingDestinationCloudwatch</summary>

```sh
$ make testacc TESTARGS='-run=TestAccHCPLogStreamingDestinationCloudwatch'
==> Checking that code complies with gofmt requirements...
golangci-lint run --config ./golangci-config.yml 
TF_ACC=1 go test ./internal/... -v -run=TestAccHCPLogStreamingDestinationCloudWatch -timeout 360m -parallel=10

=== RUN   TestAccHCPLogStreamingDestinationCloudWatch
--- PASS: TestAccHCPLogStreamingDestinationCloudWatch (6.36s)
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/provider/logstreaming      7.483s
```

</details>

[IFACE-2483]: https://hashicorp.atlassian.net/browse/IFACE-2483?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by manish-hashicorp — APPROVED** [review]

**Comment by manish-hashicorp** [comment]
Rubber stamping. Please get approval from your team before merging.

**Review by janet — APPROVED** [review]

**Review by binarymatt — APPROVED** [review]

**Pull request merged**
